File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def fibo(n):
    first = 0
    second = 1
    for i in range(n):
        first, second = second, first + second
    
    return first

def solution(n):
    
    nth_fibo = fibo(n)
    answer = nth_fibo % 1234567
    return answer

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    static int[] memo;

    public int solution(int n) {
        memo = new int[n+1];
        return fib(n);
    }

    private int fib(int n) {
        if (n == 0) return 0;
        if (n == 1) return 1;
        if (memo[n] != 0) return memo[n]; // 이미 계산된 값이면 바로 반환

        // 계산 후 저장(메모이제이션)
        memo[n] = (fib(n-1) + fib(n-2)) % 1234567;
        return memo[n];
    }
}
